app/auth/views/suppliers.py

```diff
@@ -0,0 +1,30 @@
+from flask import jsonify
+from flask_login import current_user, login_required
+from app.auth import auth
+from app.utils import get_json_from_request, json_has_required_keys
+from app.auth.suppliers import get_supplier, update_supplier_details, valid_supplier
+
+
+@auth.route('/supplier', methods=['GET'])
+@login_required
+def get_supplier_profile():
+    if not valid_supplier(current_user):
+        return jsonify(message="Only users with role supplier can access supplier details"), 401
+    else:
+        supplier = get_supplier(current_user.supplier_code)
```

Comment:
was one of the goals of this refactor to just return the fields needed for the edit form?

Reply:
This commit allows basic get and post so Marco can work on his forms. Will need to look at this